src/browser/cdp/selectivity/test-dependencies-writer.ts

```diff
@@ -0,0 +1,72 @@
+import { memoize } from "lodash";
+import path from "node:path";
+import fs from "fs-extra";
+import { shallowSortObject } from "./utils";
+import type { Test } from "../../../types";
+import type { NormalizedDependencies } from "./types";
+
+const areDepsSame = (browserDepsA?: NormalizedDependencies, browserDepsB?: NormalizedDependencies): boolean => {
+    const props: Array<keyof NormalizedDependencies> = ["js", "css", "modules"] as const;
+
+    if (!browserDepsA || !browserDepsB) {
+        return false;
+    }
+
+    for (const prop of props) {
+        if (!browserDepsA[prop] || !browserDepsB[prop] || browserDepsA[prop].length !== browserDepsB[prop].length) {
+            return false;
+        }
+    }
+
+    // Rely on the fact both are sorted arrays
+    for (const prop of props) {
+        for (let i = 0; i < browserDepsA[prop].length; i++) {
+            if (browserDepsA[prop][i] !== browserDepsB[prop][i]) {
+                return false;
+            }
+        }
+    }
+
+    return true;
+};
+
+export class TestDependenciesWriter {
+    private readonly _selectivityTestsPath: string;
+    private _directoryCreated = false;
+
+    constructor(selectivityRootPath: string) {
+        this._selectivityTestsPath = path.join(selectivityRootPath, "tests");
+    }
+
+    async saveFor(test: Test, browserDependencies: NormalizedDependencies): Promise<void> {
```

Comment:
Unlike "file-hash-writer", we dont keep dependencies for tests in memory.
Steps:
- Read
- Compare
- Write, if different
- Clean memory